### PR TITLE
Udx 6667 implement cortx pod and container layout changes

### PR DESCRIPTION
**Pull request by danthanhton**

### CODE REVIEW

**Comment by danthanhton** [comment]
This review also includes changes for UDX-6582 (modify cortx block storage helm chart to only create one storage class for all pvs and pvcs in the cluster)

**Comment by osowski** [comment]
Looks good so far to me with dummy containers. We need to work with @ujjwalpl to get the specific container arguments for the new pod layouts to function correctly.

**Review by osowski — COMMENTED** [review]
Looks good so far. We'll want to iterate with @ujjwalpl and team to ensure we have the correct container arguments before we PR this into "integration".

File: k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-ha/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+apiVersion: v1
+description: CORTX data
```
Comment (mukhtar-inamdar) [inline k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-ha/Chart.yaml:2]: Don't know but I think 
this line can be CORTX ha
and the name in below line (3) can be cortx-ha

Comment (danthanhton) [reply k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-ha/Chart.yaml:2]: Yes, thanks for catching this. Fixed.

File: k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-ha/templates/cortx-ha-pod.yaml [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+        app: {{ .Values.cortxha.name }}
+        {{- if .Values.cortxha.machineid.value }}
+        cortx.io/machine-id: {{ .Values.cortxha.machineid.value }}
+        {{- end }}
+        name: cortx-ha
+        cortx.io/service-type: cortx-ha
+    spec:
+      hostname: {{ .Values.cortxha.service.headless.name }}
+      serviceAccountName: {{ .Values.cortxha.serviceaccountname }}
+      automountServiceAccountToken: true
```
Comment (osowski) [inline k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-ha/templates/cortx-ha-pod.yaml:25]: I updated this to resolve the need for the SA token, as defined by https://jts.seagate.com/browse/UDX-6849

File: k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-ha/values.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+namespace: default
+cortxha:
+  name: cortx-ha
+  image: ghcr.io/seagate/centos:7
+  secretinfo: secret-info.txt
+  serviceaccountname: cortx-sa
```
Comment (mukhtar-inamdar) [inline k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-ha/values.yaml:6]: can define different name possible 'cortx-ha'
as for ha pod, we need to give some more permissions through clusterrole and clusterrolebinding.
it is already part of some other ticket @osowski as it is required to watch on pods for getting events.
so kipping a separate account will give this permission to HA pod only.

Comment (osowski) [reply k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-ha/values.yaml:6]: Can you create a JIRA ticket for the specific request of creating a new Service Account specific to the HA pod and the explicit permissions it requires to run with (above and beyond the rest of the basic permissions the existing CORTX SA is given)? You can assign it to me directly.

Comment (mukhtar-inamdar) [reply k8_cortx_cloud/cortx-cloud-helm-pkg/cortx-ha/values.yaml:6]: it's ok, you will get the required input from the provisioner test, team you can go by your process. :-)

**Pull request merged**